src/software/ai/hl/stp/play/enemy_free_kick/enemy_free_kick_play_fsm.cpp

```diff
@@ -0,0 +1,189 @@
+
+    auto assignments = getAllDefenderAssignments(
+        enemy_threats, event.common.world_ptr->field(), event.common.world_ptr->ball(),
+        ai_config.defense_play_config().defender_assignment_config());
+
+    if (assignments.size() == 0)
+    {
+        return;
+    }
+
+    // Adds designated free kick defender to block the direction the kicker is facing
+    if (!enemy_threats.empty())
+    {
+        auto block_free_kicker = std::make_shared<PassDefenderTactic>();
+        Vector block_direction =
+            Vector::createFromAngle(enemy_threats[0].robot.orientation());
+        block_kick_point = event.common.world_ptr->ball().position() +
+                           block_direction.normalize(0.6 + 2 * ROBOT_MAX_RADIUS_METERS);
+        block_free_kicker->updateControlParams(block_kick_point);
+        tactics_to_return[0].push_back(block_free_kicker);
+    }
+
+    // Choose which defender assignments to assign defenders to based on number
+    // of tactics available to set after assigning kick off defender
+    std::vector<DefenderAssignment> crease_defender_assignments;
+    std::vector<DefenderAssignment> pass_defender_assignments;
+    unsigned int assigns_skipped = 0;
+    for (unsigned int i = 0; i < num_defenders; i++)
+    {
+        DefenderAssignment defender_assignment;
+        if (i < assignments.size())
+        {
+            defender_assignment = assignments.at(i + assigns_skipped);
+            defender_assignment = assignments.at(i);
+
+            while (defender_assignment.type == PASS_DEFENDER &&
+                   distance(defender_assignment.target, block_kick_point) <= 0.5)
+            {
+                if (i + assigns_skipped + 1 >= assignments.size())
+                {
+                    defender_assignment = assignments.front();
+                    break;
+                }
+                assigns_skipped++;
+                defender_assignment = assignments.at(i + assigns_skipped);
+            }
+        }
+        else
+        {
+            // If we have more tactics to set than determined defender assignments,
+            // assign remaining defenders to the defender assignment with the
+            // highest coverage rating
+            defender_assignment = assignments.front();
+        }
+
+        if (defender_assignment.type == CREASE_DEFENDER)
+        {
+            crease_defender_assignments.emplace_back(defender_assignment);
+
+            // If we have at least two available defenders, two defenders should
+            // be assigned to the highest scoring crease defender assignment to better
+            // block the shot cone of the most threatening enemy
+            if (i == 0 && num_defenders >= 2)
+            {
+                crease_defender_assignments.emplace_back(defender_assignment);
+                i++;
+            }
+        }
+        else
+        {
+            pass_defender_assignments.emplace_back(defender_assignment);
+        }
+    }
```

Comment:
if we have extra defenders, we could also place a pass defender between the enemy free kicker and our net IF the first pass defender is far away from that position